### PR TITLE
[DOC] Self-SL & SupCon for classification

### PR DESCRIPTION
**Pull request by sungchul1**
### Self-SL
- [x] multi-class classification
- [ ] ~multi-label classification~
-> It is required to get performance, so I'll update it in another PR.

### SupCon
- [x] multi-class classification
- [ ] ~hierarchical classification~

### CODE REVIEW

**Comment by sungchul1** [comment]
## multi-class classification
### Training technique & SupCon
image

### Self-SL
image

**Comment by sungchul2** [comment]
### hierarchical classification
image

**Review by harimkang — APPROVED** [review]
I think we can omit this integration test and merge. @goodsong81

**Review by goodsong81 — APPROVED** [review]

File: docs/source/explanation/algorithms/classification/multi_class_classification.rst [hunk 1]
```diff
@@ -14,10 +14,42 @@ For the supervised training we use the following algorithms components:
...
+            --learning_parameters.early_stop_iteration_patience=8 \ # (for iter runner) stop if the model don't improve within the number of iterations of patience
+
+    - `Balanced Sampler <https://github.dev/openvinotoolkit/training_extensions/blob/develop/otx/mpa/modules/datasets/samplers/balanced_sampler.py#L11>`_: To create an efficient batch that consists of balanced samples over classes, reducing the iteration size as well.
+    - `Supervised Contrastive Learning (SupCon) <https://arxiv.org/abs/2004.11362>`_: To enhance the performance of the algorithm in case when we have a small number of data. More specifically, we train a model with two heads: classification head with Influence-Balanced Loss and contrastive head with `Barlow Twins loss <https://arxiv.org/abs/2103.03230>`_. It enables using `--learning_parameters.enable_supcon=True` in CLI.
+      The below table shows how much performance SupCon improved compared with baseline performance on three baseline datasets with 10 samples per class: CIFAR10, Eurosat-10, and Food-101.
+
+      +-----------------------+---------+---------+------------+---------+----------+---------+
```
Comment (harimkang) [inline docs/source/explanation/algorithms/classification/multi_class_classification.rst:34]: @sungchul2 I found it while checking the latest built documentation. Is this table location right here? It is marked as Self-SL.

Comment (sungchul2) [reply docs/source/explanation/algorithms/classification/multi_class_classification.rst:34]: I updated it at https://github.com/openvinotoolkit/training_extensions/pull/1756.
Could you check it?

Comment (sungchul1) [reply docs/source/explanation/algorithms/classification/multi_class_classification.rst:34]: I updated table columns at https://github.com/openvinotoolkit/training_extensions/pull/1756.
Could you check it?

**Pull request merged**